app/AccountancyModule/Grids/DtoListDataSource.php

```diff
@@ -29,6 +33,8 @@ public function __construct(array $data)
 
     /**
      * @param mixed[] $filters
+     *
+     * @phpstan-return $this
```

Comment:
to je k čemu?

Reply:
Třída `DtoListDataSource` je totiž generická takže pokud voláš `(DtoListDataSource<T>()).filter([])`, tak ti nestačí return type `DtoListDataSource`, protože z něj není jasný ten typ `T`.

Tím `$this` říkáš, že se vrací stejná instance, jako ta na které tu metodu voláš. Alternativou by bylo mít jako return type `DtoListDataSource<T>`. Tohle je o něco striktnější a plní to stejnou funkci, tak jsem tím směrem šel.

Reply:
A `self` co je uveden jako návratová hodnota jim nestačí?

Reply:
Nestačí.

Uvažuj třeba metodu Collection::map(), ta určitě vrací self, ale může mapovat ty hodnoty na jinej typ. Tohle je potřeba explicitně uvést.